### PR TITLE
[DOCS] How-to guide: How to configure a BigQuery Datasource

### PR DESCRIPTION
**Pull request by eugmandel**
Changes proposed in this pull request:
- How-to guide: How to configure a BigQuery Datasource (following the new how-to guide template)

Previous Design Review notes:
- The steps are copied from the older version and were tested back then.

### CODE REVIEW

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 1]
```diff
@@ -1,59 +1,61 @@
...
+
+2. Choose "Big Query" from the list of database engines, when prompted.
+3. Consult the `PyBigQuery <https://github.com/mxmzdlv/pybigquery`_ docs
+   for help building a connection string for your BigQuery cluster.
+
+    If you want GE to connect to your BigQuery project (without specifying a particular dataset), the URL should be:
 
     .. code-block:: python
 
         "bigquery://project-name"
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]: No quotation marks, right?

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 2]
```diff
@@ -1,59 +1,61 @@
...
+
+  - :ref:`Set up a working deployment of Great Expectations <getting_started>`
+  - Followed the `Google Cloud library guide <https://googleapis.dev/python/google-api-core/latest/auth.html>`_ for authentication
+  - Installed the pybigquery package for the BigQuery sqlalchemy dialect (``pip install pybigquery``)
+
+Steps
+-----
+
+
+1. First, run this CLI command:
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
1. Run the following CLI command to begin the interactive Datasource creation process:
```

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 3]
```diff
@@ -1,59 +1,61 @@
...
+
+.. code-block:: bash
+
+    great_expectations datasource new
+
+
+2. Choose "Big Query" from the list of database engines, when prompted.
+3. Consult the `PyBigQuery <https://github.com/mxmzdlv/pybigquery`_ docs
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
3. Identify the connection string you would like Great Expectations to use to connect to BigQuery, using the examples below and the `PyBigQuery <https://github.com/mxmzdlv/pybigquery>`_ documentation.

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 4]
```diff
@@ -1,59 +1,61 @@
...
+   for help building a connection string for your BigQuery cluster.
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 5]
```diff
@@ -1,59 +1,61 @@
...
+
+    If you want GE to connect to your BigQuery project (without specifying a particular dataset), the URL should be:
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
    If you want Great Expectations to connect to your BigQuery project (without specifying a particular dataset), the URL should be:
```

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 6]
```diff
@@ -1,59 +1,61 @@
...
 
     .. code-block:: python
 
         "bigquery://project-name"
 
 
-5. Paste in this connection string and finish out the interactive prompts.
-6. Should you need to modify your connection string you can manually edit the
-   ``great_expectations/uncommitted/config_variables.yml`` file.
+    If you want GE to connect to a particular dataset inside your BigQuery project, the URL should be:
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
    If you want Great Expectations to connect to a particular dataset inside your BigQuery project, the URL should be:
```

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 7]
```diff
@@ -1,59 +1,61 @@
...
-Users can specify a table via a Batch Kwarg called ``bigquery_temp_table``:
+        "bigquery://project-name/dataset-name"
 
-    .. code-block:: python
 
-        batch_kwargs = {
-            "query": "SELECT * FROM `my-project.my_dataset.my_table`",
-            "bigquery_temp_table": "my_other_dataset.temp_table"
-        }
+    If you want GE to connect to one of the Google's public datasets, the URL should be:
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
    If you want Great Expectations to connect to one of the Google's public datasets, the URL should be:
```

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 8]
```diff
@@ -1,59 +1,61 @@
...
-If a default BigQuery dataset is defined in the connection string
-(for example, ``bigquery://project-name/dataset-name``), and no ``bigquery_temp_table``
-Batch Kwarg is supplied, then GE will create a permanent table with a random
-UUID in that location (e.g. ``project-name.dataset-name.ge_tmp_1a1b6511_03e6_4e18_a1b2_d85f9e9045c3``).
+        "bigquery://project-name/bigquery-public-data"
 
-If a default BigQuery dataset is not defined in the connection string
-(for example, ``bigquery://project-name``) and no ``bigquery_temp_table`` Batch Kwawrg
-is supplied, then custom queries will fail.
+5. Paste in this connection string when prompted for SQLAlchemy URL and finish out the interactive prompts.
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
5. Enter the connection string when prompted, and finish completing the interactive prompts.
```

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 9]
```diff
@@ -1,59 +1,61 @@
...
+6. Should you need to modify your connection string you can manually edit the
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]:
```suggestion
6. Should you need to modify your connection string, you can manually edit the
```

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst [hunk 10]
```diff
@@ -1,59 +1,61 @@
...
+   ``great_expectations/uncommitted/config_variables.yml`` file.
 
 
 Additional notes
 ----------------
 
-Follow the `Google Cloud library guide <https://googleapis.dev/python/google-api-core/latest/auth.html>`_
-for authentication.
+Environment variables can be used to store the SQLAlchemy URL instead of the file, if preferred - search documentation for "Managing Environment and Secrets".
```
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst:57]: Can we provide a link?

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst (hunk 8 above)
Comment (jcampbell) [inline docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]: What additional interactive prompts are coming?

**Review by jcampbell — CHANGES_REQUESTED** [review]
Comments and suggestions inline

**Review by jcampbell — APPROVED** [review]

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst (hunk 10 above)
Comment (eugmandel) [reply docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst:57]: Done

File: docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst (hunk 8 above)
Comment (eugmandel) [reply docs/how_to_guides/configuring_datasources/how_to_configure_a_bigquery_datasource.rst]: Made it more concrete.

**Pull request merged**